saleor/order/models.py

```diff
@@ -323,7 +325,7 @@ class Order(ModelWithMetadata, ModelWithExternalReference):
     objects = OrderManager()
 
     class Meta:
-        ordering = ("-number",)
+        ordering = ("-updated_at",)
```

Comment:
Now we accept any string as number values, so I believe ordering by number_as_str has no sense anymore.